pyproject.toml

```diff
@@ -0,0 +1,114 @@
+[build-system]
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[tool.hatch.version]
+path = "omc3/__init__.py"
+
+[tool.hatch.build.targets.sdist]
+include = [
+  "/omc3",
+]
+exclude = [
+  "/.github",
+  "/doc",
+  "/tests",
+]
+
+[tool.hatch.build.targets.wheel]
+packages = ["omc3"]
+
+[project]
+name = "omc3"
+readme = "README.md"
+description = "An accelerator physics tools package for the OMC team at CERN."
+authors = [
+    {name = "OMC Team", email = "pylhc@github.com"},  # see zenodo file / commits for details
+]
+license = "MIT"
+dynamic = ["version"]
+requires-python = ">=3.9"
+
+classifiers = [
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: MIT License",
+    "Natural Language :: English",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Topic :: Scientific/Engineering :: Physics",
+    "Topic :: Scientific/Engineering :: Visualization",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+    "Typing :: Typed",
+]
+
+dependencies = [
+  "numpy >= 1.24",
+  "scipy >= 1.10",
+  "pandas >= 2.1",
+  "tfs-pandas >= 3.8",
+  "matplotlib >= 3.8",
+  "Pillow >= 6.2.2",  # not our dependency but older versions crash with mpl
+  "generic-parser >= 1.1",
+  "sdds >= 0.4",
+  "optics-functions >= 0.1",
+  "turn_by_turn >= 0.6",
+  "uncertainties >= 3.1",
+  "scikit-learn >= 1.0",
+  "h5py >= 2.9",
+  "tables >= 3.6",
```

Comment:
anything we can do about this?

Reply:
That would require a rework of the response matrices output, which is where we use it. Currently we have an HDF table storing the matrices but we might want a zipped folder with one .parquet per matrix for instance. 

I think it's out of this PR's scope though.

Reply:
I meant mostly for the tests and for allowing numpy 2.0 for now

Reply:
I have already excluded the relevant tests from the run, but the problem is things fail at import time. As soon as we import omc3 somewhere down the line there is an `import tables` statement and that crashes the run :(

Reply:
Setting numpy<2 for now